qemu/tests/balloon_boot_in_pause.py

```diff
@@ -0,0 +1,233 @@
+import logging
+import random
+
+from avocado.core import exceptions
+
+from virttest import error_context
+from virttest import utils_misc
+from qemu.tests.balloon_check import BallooningTest
+
+
+class BallooningTestPause(BallooningTest):
+
+    """
+    Basic functions of memory ballooning test for guest booted
+    in paused status
+    """
+
+    def __init__(self, test, params, env):
+        super(BallooningTest, self).__init__(test, params, env)
+
+        self.vm = env.get_vm(params["main_vm"])
+        self.ori_mem = self.get_vm_mem(self.vm)
+        if self.get_ballooned_memory() != self.ori_mem:
+            self.balloon_memory(self.ori_mem)
+        self.old_mmem = self.ori_mem
+        self.old_gmem = None
+
+    @error_context.context_aware
+    def memory_check(self, step, changed_mem):
+        """
+        Check memory change status in monitor and return memory both
+        in guest and monitor
+        :param step: the check point string
+        :type step: string
+        :param changed_mem: ballooned memory in current step(compared with
+        last round of memory, i.e. self.old_mmem, instead of self.ori_mem)
+        :type changed_mem: int
+        :return: memory size get from monitor and guest
+        :rtype: tuple
+        """
+        error_context.context("Check memory status %s" % step, logging.info)
+        mmem = self.get_ballooned_memory()
+        gmem = self.get_memory_status()
+        if (abs(mmem - self.old_mmem)) != changed_mem or (
+                    self.old_gmem and (
+                            abs(gmem - self.old_gmem) - changed_mem) > 100):
+            self.error_report(step, abs(self.old_mmem - changed_mem),
+                              mmem, gmem)
+            raise exceptions.TestFail("Balloon test failed %s" % step)
+        return (mmem, gmem)
+
+    @error_context.context_aware
+    def balloon_memory(self, new_mem):
+        """
+        Baloon guest memory to new_mem
+
+        :param new_mem: New desired memory.
+        :type new_mem: int
+        """
+        error_context.context("Change VM memory to %s" % new_mem, logging.info)
+        try:
+            self.vm.balloon(new_mem)
+        except Exception, e:
+            if self.vm.monitor.verify_status('paused'):
+                # Make sure memory not changed before the guest resumed
+                if self.get_ballooned_memory() != self.ori_mem:
+                    raise exceptions.TestFail("Memory changed before guest "
```

Comment:
it's better to add the blank in the next line, it's not good habit to add it at the end of line.